web_frontend/cloudscheduler/glintwebui/glint_utils.py

```diff
@@ -195,21 +203,90 @@ def get_image(config, image_name, image_checksum, group_name, cloud_name=None):
         if len(image_candidates) > 0:
             return image_candidates[0]
         else:
-            #No image that fits specs
+            # No image that fits specs
             return False
 
 
 # at a length of 16 with a 85 symbol alphabet we have a N/16^85 chance of a collision, pretty darn unlikely
 def generate_tx_id(length=16):
     return ''.join(random.choice(ALPHABET) for i in range(length))
 
+
+def convert_sparsify_compress(src_file_path, virt_sparsify, with_compression):
+    if os.path.exists(src_file_path):
+        print("process starts")
+        added_cmd = ''
+        is_sparsified = ''
+        is_compressed = ''
+        # report_message = ''
+
+        if virt_sparsify:
+            print("Virt sparsify checkbox is selected")
+
+            # make a copy of original file
+            backup_copy_file_path = src_file_path + ".cp"
+            sub_command = "cp -p %s %s" % (src_file_path, backup_copy_file_path)
+            os.system(sub_command)
+
+            # virt-sparsify the source file and see if there's any error or warning
+            sub_command = "virt-sparsify --in-place %s" % src_file_path
+            output = os.popen(sub_command).read()
+            print("\n THE POPEN OUTPUT IS:\n" + output + "\nTHE POPEN OUTPUT ENDS\n")
+
+            # if there's any warning or error in virt-sparsify, make the copy file be the source file
+            if len(output) == 0 or "warning" in output:
+                print("Can't virt-sparsify")
+                sub_command = "rm %s" % src_file_path
+                os.system(sub_command)
+
+                sub_command = "mv %s %s" % (backup_copy_file_path, src_file_path)
+                os.system(sub_command)
+
+                # if there's error in virt-sparsify
+                if len(output) == 0:
+                    print("virt-sparsify: Error")
+                    # report_message = "virt-sparsify: error: libguestfs error: discard cannot be enabled on this drive: " \
+                    #                  "qemu does not support discard for files in this disk format. "
+
+                # if there's warning in virt-sparsify
+                else:
+                    print("virt-sparsify: Warning")
+                    # report_message = "virt-sparsify: warning: fstrim operation is not supported on /dev/sda1 (vfat). "
+            else:
+                # if virt-sparsify goes well, delete the copy file to save some space
+                print("virt-sparsify: successfully")
+                is_sparsified = '.reorganized'
+
+                sub_command = "rm %s" % backup_copy_file_path
+                os.system(sub_command)
+
+        if with_compression:
+            print("With compression checkbox is selected")
+            added_cmd = "-c"
+            is_compressed = ".compressed"
+
+        output_format = 'qcow2'
+        dest_file_path = src_file_path + is_sparsified + is_compressed + ".qcow2"
```

Comment:
I believe in our most recent discussion we decided not to add any additional suffix during these processes so csv2 can treat them as the same image.